python/huggingfaceserver/huggingfaceserver/vllm/vllm_completions.py

```diff
@@ -388,13 +388,17 @@ def _validate_input(
             request.max_tokens = self.max_model_len - token_num
 
         if token_num + request.max_tokens > self.max_model_len:
-            raise InvalidInput(
```

Comment:
https://github.com/kserve/kserve/blob/e82beb6c4cdac44d15ebc900d49d1be947861d01/python/kserve/kserve/protocol/rest/server.py#L88

hmm InvalidInput is supposed to return 400
https://github.com/kserve/kserve/blob/e82beb6c4cdac44d15ebc900d49d1be947861d01/python/kserve/kserve/errors.py#L110

Reply:
That exception is being caught here and getting converted to OpenAIError: https://github.com/kserve/kserve/blob/7dd4ae2e5c6c821a4a3b875ffcc35a70b18a3234/python/huggingfaceserver/huggingfaceserver/vllm/vllm_completions.py#L162

Even if we return it as InvalidInput error, the format will be in kserve error format which is inconsistent with the rest of the errors returned here. I mentioned the difference in an issue I created: https://github.com/kserve/kserve/issues/3887

Reply:
Gotcha, for openai completion and chat completion endpoints we should return the OpenAIError. We will check the case for huggingface backend.